src/model.rs

```diff
@@ -89,36 +88,11 @@ pub struct Transaction {
 pub enum TransactionData {
     Stake(Staker), // FIXME: to remove, this is temporary waiting for real staking contract !!
     Blob(BlobTransaction),
-    Proof(ProofTransaction),
-    VerifiedProof(VerifiedProofTransaction),
-    RecursiveProof(RecursiveProofTransaction),
-    VerifiedRecursiveProof(VerifiedRecursiveProofTransaction),
+    MultiProof(MultiProofTransaction),
+    VerifiedMultiProof(VerifiedMultiProofTransaction),
```

Comment:
If we have no "singleProof" here, but only "Mutli" proof, then I would keep the type names & just change ProofTransaction to accept a Vec instead of a single element

at the beginning, ProofTransaction had a vec of BlobReference that we dropped to simplify the process, we can re-rollback on it and re-add this vec